### PR TITLE
[frontend/backend] Apply marking definition to export files (#5797)

### LINKED ISSUE #5797: Apply marking definition to export files
## Use case

Example: When I import or export files containing entities with a TLP:RED, I would like that other users with export and import rights, but not TLP:RED rights, not to be able to see my file.

Temporary work created to make the file export "spin" before the real export must also be protected

### PR DESCRIPTION
**Pull request by Goumies**
### Proposed changes

* Update InternalFile entity model with content, file, and object markings
* Update export generation forms with required fields: content and file max definition levels
* Add markings to API and export report pdf connector communications

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/5797

### Checklist

- [ ] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

/!\ For this feature to be validated, evolutions in  `Client-Python` and `Connectors` must be reviewed and merged first. /!\ 
This pull request is related to the ones in the `Client-Python` and `Connectors` :
* https://github.com/OpenCTI-Platform/client-python/pull/613
* https://github.com/OpenCTI-Platform/connectors/pull/2011

### CODE REVIEW

**Comment by Goumies** [comment]
The current work submitted in this Pull Request fulfills 1 / 3 acceptance criteria.
Please take a look at the brainstorming for more details.

We have to update the export list in the frontend to finalize this feature :
- The `exportFiles` pagination only contains the id of every file. Despite the spread fragment presence. The idea was to compare the current user markings and the file ones for every file. Julien has mentioned a mechanism to control the access to data with the markings. It is supposed to automatically compare the resolved allowed markings of the user and the markings of the data to file in the current location.
- The initial approach to store the Internal File in Elastic was to replace the manual file indexing with the usual Entity creation: _indexFileToDocument_ -> createEntity. I kept both because `createEntity` alone was breaking the exportFiles pagination in the frontend. And, importing createEntity in file storage provokes a dependency cycle between `file-storage.js` and `middleware.js`

image 

I have left comments in the code source where I made some attempts.

/!\ `maxMarking` (string) is now renamed to `contentMarkings` or `contentMaxMarkings` (array) : 
impact of this change still needs to be tested.

File: opencti-platform/opencti-front/src/components/AutocompleteField.jsx [hunk 1]
```diff
@@ -80,7 +80,7 @@ const AutocompleteField = (props) => {
             name={name}
             fullWidth={true}
             error={!isNil(meta.error) && meta.touched}
-            helperText={meta.error || textfieldprops.helperText}
+            helperText={(meta.touched && meta.error) || textfieldprops.helperText}
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/components/AutocompleteField.jsx:83]: what does this change do ?

Comment (CelineSebe) [reply opencti-platform/opencti-front/src/components/AutocompleteField.jsx:83]: This prevents the error message from appearing under the 'object markings definition' input

**Comment by Archidoit** [comment]
When I export the list of reports in txt, I get a file with this content : 
image

**Comment by Archidoit** [comment]
When I try to generate an export without 'file marking definition level', I get no error and nothing happens when I click on 'create' ( --> an error should be indicated): 
image

Whereas trying to create it without 'content max marking definition levels' indicates an error : 
image

**Comment by Archidoit** [comment]
In the overview of a report, I click on the 'export button' and generate a txt export. The file appears in the 'data' tab with an error (because txt only supports list exports).
But then, when I click on an other tab, an error occurs
image

**Comment by Archidoit** [comment]
When generating the export of an entity in txt with these markings : 
image

I get an error at file export : 
image

**Comment by Archidoit** [comment]
Generate the export of an attack pattern, after clicking on 'create', we are redirected towards this page (instead of the 'Data' tab) : 
image

**Comment by CelineSebe** [comment]
> When I try to generate an export without 'file marking definition level', I get no error and nothing happens when I click on 'create' ( --> an error should be indicated): image
> 
> Whereas trying to create it without 'content max marking definition levels' indicates an error : image

An error is only indicated when the input is touched for both, not before. 
This behavior appears to be consistent. @nino-filigran
But  when I try to generate an export without 'file marking definition level', I get error on my side.

**Comment by CelineSebe** [comment]
> Generate the export of an attack pattern, after clicking on 'create', we are redirected towards this page (instead of the 'Data' tab) : image

This bug has been resolved on master.

**Comment by Archidoit** [comment]
> > Generate the export of an attack pattern, after clicking on 'create', we are redirected towards this page (instead of the 'Data' tab) : image
> 
> This bug has been resolved on master.

No, i still have the bug on master (for attack patterns)

**Comment by JeremyCloarec** [comment]
Csv export connector doesn't seem to take content max markings into account. 
I exported a report with marking TLP:AMBER, 
image
using TLP:GREEN as content max markings
image
but the exported file contains the report and the observable in it:
image

This doesn't occur with stix export connector

EDIT: this bug is already occuring on testing

--> created an issue: https://github.com/OpenCTI-Platform/opencti/issues/6745

**Comment by Archidoit** [comment]
Issue with observables export

Go to the observables tab. Generate an export. The pop-up only contains 'max markings': 
image

and an error occurs at export: 
image

**Comment by Archidoit** [comment]
Go to a feedback overview. Generate a csv simple export (ie only the entity).
You are redirected to (instead of the Data tab) : 
image


And if you go to the Data tab, the generated file contains both the feedback and its related entities : 
image


--> create an issue : https://github.com/OpenCTI-Platform/opencti/issues/6746

File: opencti-platform/opencti-front/src/private/components/common/files/FileLine.tsx [hunk 2]
```diff
@@ -497,7 +497,7 @@ const FileLineComponent: FunctionComponent<FileLineComponentProps> = ({
   );
 };
 
-const FileLine = createFragmentContainer(FileLineComponent, {
+const FileLine = createFragmentContainer(FileLineComponent, { // objectMarking -> rel_object-marking.internal_id
```
Comment (JeremyCloarec) [inline opencti-platform/opencti-front/src/private/components/common/files/FileLine.tsx]: I don't get what this comment means

File: opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectFileExport.tsx [hunk 3]
```diff
@@ -75,17 +81,23 @@ const StixCoreObjectFileExportComponent = ({
...
   const onSubmitExport = (
     values: FormValues,
-    { setSubmitting, resetForm }: FormikHelpers<FormValues>,
+    { setSubmitting, setErrors, resetForm }: FormikHelpers<FormValues>,
   ) => {
-    const maxMarkingDefinition = values.maxMarkingDefinition === 'none'
-      ? null
-      : values.maxMarkingDefinition;
+    const contentMaxMarkingDefinitions = values.contentMaxMarkingDefinitions.map(({ value }) => value); // rename to contentMaxMarkings
+    const fileMarkingDefinitions = values.fileMarkingDefinitions.map(({ value }) => value); // rename to fileMarkingDefinitions
```
Comment (JeremyCloarec) [inline opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectFileExport.tsx]: comments should be removed

File: opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectFilesAndHistory.jsx [hunk 4]
```diff
@@ -95,7 +89,7 @@ export const stixCoreObjectFilesAndHistoryExportMutation = graphql`
       }
     }
   }
-`;
+`; // retrieve objectMarking or rel_object-marking.internal_id
```
Comment (JeremyCloarec) [inline opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectFilesAndHistory.jsx]: I don't get this comment either

File: opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectsExportCreation.jsx [hunk 5]
```diff
@@ -112,15 +116,17 @@ class StixDomainObjectsExportCreationComponent extends Component {
 
   onSubmit(selectedIds, values, { setSubmitting, resetForm }) {
     const { paginationOptions, exportContext } = this.props;
-    const maxMarkingDefinition = values.maxMarkingDefinition === 'none'
-      ? null
-      : values.maxMarkingDefinition;
+
+    const contentMaxMarkingDefinitions = values.contentMaxMarkingDefinitions.map(({ value }) => value); // rename to contentMaxMarkings
+    const fileMarkingDefinitions = values.fileMarkingDefinitions.map(({ value }) => value); // rename to fileMarkings
```
Comment (JeremyCloarec) [inline opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectsExportCreation.jsx]: comments to remove

Comment (CelineSebe) [reply opencti-platform/opencti-front/src/private/components/common/stix_domain_objects/StixDomainObjectsExportCreation.jsx]: Done

File: opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectFileExport.tsx (hunk 3 above)
Comment (CelineSebe) [reply opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectFileExport.tsx]: Done

File: opencti-platform/opencti-graphql/src/schema/stixMetaObject.ts [hunk 6]
```diff
@@ -3,6 +3,7 @@ import { schemaTypesDefinition } from './schema-types';
 
 export const ENTITY_TYPE_LABEL = 'Label';
 export const ENTITY_TYPE_EXTERNAL_REFERENCE = 'External-Reference';
+export const ENTITY_TYPE_INTERNAL_FILE = 'Internal-File';
```
Comment (JeremyCloarec) [inline opencti-platform/opencti-graphql/src/schema/stixMetaObject.ts:6]: we already have an ENTITY_TYPE_INTERNAL_FILE in schema/internalObject, is there a reason to duplicate it?

File: opencti-platform/opencti-graphql/src/modules/relationsRef/stixCoreObject-registrationRef.ts [hunk 7]
```diff
@@ -1,8 +1,8 @@
 import { schemaRelationsRefDefinition } from '../../schema/schema-relationsRef';
 import { ABSTRACT_STIX_CORE_OBJECT } from '../../schema/general';
-import { createdBy, externalReferences, objectLabel, objectMarking, xOpenctiLinkedTo } from '../../schema/stixRefRelationship';
+import { createdBy, externalReferences, internalFiles, objectLabel, objectMarking, work, xOpenctiLinkedTo } from '../../schema/stixRefRelationship';
 
 schemaRelationsRefDefinition.registerRelationsRef(
   ABSTRACT_STIX_CORE_OBJECT,
-  [createdBy, objectMarking, objectLabel, externalReferences, xOpenctiLinkedTo]
+  [createdBy, objectMarking, objectLabel, externalReferences, internalFiles, work, xOpenctiLinkedTo]
```
Comment (JeremyCloarec) [inline opencti-platform/opencti-graphql/src/modules/relationsRef/stixCoreObject-registrationRef.ts]: i don't get these new refs for stix core objects. When and how are they used? Is it in preparation of a future chunk?

File: opencti-platform/opencti-front/src/private/components/common/files/FileLine.tsx (hunk 2 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/private/components/common/files/FileLine.tsx]: It was meant to tell me how to implement the marking relation.
I just forgot to remove it.

Thanks.

File: opencti-platform/opencti-graphql/src/modules/relationsRef/stixCoreObject-registrationRef.ts (hunk 7 above)
Comment (Goumies) [reply opencti-platform/opencti-graphql/src/modules/relationsRef/stixCoreObject-registrationRef.ts]: Those refs indicate the possible relations.
Adding them allowed us to add markings to those entities.

File: opencti-platform/opencti-graphql/src/schema/stixMetaObject.ts (hunk 6 above)
Comment (Goumies) [reply opencti-platform/opencti-graphql/src/schema/stixMetaObject.ts:6]: No, we had no idea at that time.
Thanks.

File: opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectFilesAndHistory.jsx (hunk 4 above)
Comment (Goumies) [reply opencti-platform/opencti-front/src/private/components/common/stix_core_objects/StixCoreObjectFilesAndHistory.jsx]: Another oversight

File: opencti-platform/opencti-graphql/src/modules/relationsRef/stixCoreObject-registrationRef.ts (hunk 7 above)
Comment (Goumies) [reply opencti-platform/opencti-graphql/src/modules/relationsRef/stixCoreObject-registrationRef.ts]: On second thought, those refs are not needed
I remove them

**Review by ValentinBouzinFiligran — APPROVED** [review]

**Pull request merged**